src/background/user-stored-data-cleaner.ts

```diff
@@ -1,20 +1,16 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import * as _ from 'lodash/index';
+import { each } from 'lodash';
 
-import { BrowserAdapter } from './browser-adapter';
+import { StorageAPI } from './browser-adapters/storage-adapter';
 
 export class UserStoredDataCleaner {
-    private _browserAdapter: BrowserAdapter;
-
-    constructor(adapter: BrowserAdapter) {
-        this._browserAdapter = adapter;
-    }
+    constructor(private readonly storageAdapter?: StorageAPI) {}
```

Comment:
I'm curious why we want `storageAdapter` to be optional here, since `cleanUserData` will use it.

Reply:
Good catch!